GTC-3085: Add Admin Version to Area Representation

gtempus opened the pull request:
[Miro](https://miro.com/app/board/uXjVLKsI-90=/?moveToWidget=3458764609724767653&cot=14)

[GTC-3085](https://gfw.atlassian.net/browse/GTC-3085)

This PR adds the administrative source information to the representation of a user's administrative boundary Area. 
It does NOT alter a custom Area.

Currently, an administrative boundary Area has information in either its `iso` or `admin` property:
```js
{
  iso: {
    country: 'HND',
    region: '8'
  },
  admin: {
    adm0: 'HND',
    adm1: '8'
  }
}
```

Implicitly, the IDs refer to specific versions of GADM IDs.

This PR makes the specific version **explicit**:
```js
{
  iso: {
    country: 'HND',
    region: '8',
    source: {
      provider: 'gadm',
      version: '3.6'
    }
  },
  admin: {
    adm0: 'HND',
    adm1: '8',
    source: {
      provider: 'gadm',
      version: '3.6'
    }
  }
}
```
----

# Test
command line:
```shell
clear; NODE_PATH=./src npx mocha test/unit/**/*.test.js
```

docker:
```shell
./area.sh test
```

## Behaviors Tested in this PR
```shell
  adminSourceUtils
    Adding GADM 2.8 Source Information
      ✔ should add GADM 2.8 source information
      An Area That Is NOT An Administrative Boundary
        ✔ should not add source information to the iso attribute
    Adding GADM 3.6 Source Information
      ✔ should add GADM 3.6 source information
      An Administrative Boundary Area with Both an "iso" and "admin" Property
        ✔ should NOT add source information to an empty "iso" object
      An Area That Is NOT An Administrative Boundary
        ✔ should not add source information to the iso attribute
        ✔ should not add source information to the admin attribute

  Area Serializer
    ✔ should include the name of the Area
    ✔ should include the geostore of the Area
    Administrative Boundary IDs
      The ISO attribute
        ✔ should include the country and region
        ✔ should include the administrative ID provider and version
    An Area That Is NOT An Administrative Boundary
      ✔ should not add source information to the iso attribute

  Area Serializer V2
    ✔ should include the name of the Area
    ✔ should include the geostore of the Area
    Administrative Boundary IDs
      The ISO attribute
        ✔ should include the country, region, and subregion
        ✔ should include the administrative ID provider and version
      The Admin attribute
        ✔ should include the adm0, adm1, and adm2 IDs
        ✔ should include the administrative ID provider and version
      An Area That Is NOT An Administrative Boundary
        ✔ should not add source information to the iso attribute
        ✔ should not add source information to the admin attribute

```

gtempus committed:
:white_check_mark: test(SerializerV2): add initial tests for admin boundary areas

GTC-3085

gtempus committed:
feat(SerializerV2): add `source` attribute to representations

For administrative boundary areas, add a `source` attribute that describes the provider and the version of the IDs being used.

GTC-3085

gtempus committed:
refactor(SerializerV2): extract addSource method

gtempus committed:
refactor(SerializerV2): use `null` instead of `undefined`

gtempus committed:
:white_check_mark: test(Serializer): add initial tests for admin boundary areas

GTC-3085

gtempus committed:
refactor: moved behavior for adding `source` to its own module

GTC-3085

gtempus committed:
refactor(adminSourceUtils): clean up the check for admin area

GTC-3085

gtempus committed:
:white_check_mark: test: do not add `source` info to custom areas

GTC-3085

gtempus committed:
:broom: chore: update build configs for development

jterry64 reviewed (approved):
I think this looks good, but did challenge my understanding of the design, so wanted to make sure I understand:

Here, you're just changing the representation, and so you're returning a particular GADM version based on the areas service version, since that's how we currently do it. And currently, these two versions already have areas that point to the appropriate geostore?

Then, in a different PR, you're going to actually add to the area model to allow for multiple GADM versions on on one area?

gtempus commented:
Hi, @jterry64! 👋 
I just wanted to make explicit what is already implicit in this service (although the docs do state that V1 is `2.8` and V2 is `3.6`. We'll update them as we roll out the changes). You are absolutely right; an Area's _model_ will allow for multiple GADM versions.

# For this PR:
## Benefits to Flagship Development
Earliest integration possible. This change will give Flagship devs a concrete example of how they can detect older versions of GADM usage. 

## Benefits to Data API
We're taking small bites. Small [survivable changes](https://tidyfirst.substack.com/p/thinkie-survivable) that can be verified and delivered.

Kent's post is paywalled. But the gist is that focusing on small, survivable lower-level changes makes the higher-level (GADM update) reversible (low risk). We take it one little step at a time.

CC @dmannarino @tanderegg

gtempus committed:
feat(Serializers): ensure that representation updates stick

I found out that mongo models, when serialized, don't act like POJOs. Adding a property to an existing object wasn't working as expected. Converting to a POJO before doing the admin updates work. This caused several of the end-to-end tests to finally start failing. That was a good thing!

gtempus committed:
feat: do not add source to empty 'iso' properties of an Area

gtempus committed:
refactor: create the pojo areas in the serializers
